docs/guide/plugins.md

```diff
@@ -0,0 +1,202 @@
+---
+sidebar: auto
+---
+
+# Plugins
+
+## Creating plugin
+
+Plugin development can be easily made locally. First, we need to choose where would we want to plug into. To see all available plugin slots we may want to select `yes` during `shopware-pwa init` command when it asks to allow dev mode.
+
+![cli plugin dev mode](./../assets/cli-plugin-dev-mode.png)
+
+It shows then a button in the footer, which allows us to turn on Dev mode, and see all plugin slots in theme.
+
+![turn-on-dev-mode](./../assets/turn-on-dev-mode.png)
+
+it will show us available slots on the theme, and you can click on any of them to see its name in the console
+
+![theme-slots](./../assets/theme-slots.png)
+
+### Local plugin
+
+Let's create `cool-promotion-plugin`, which can display our special product under the top header.
+We're creating following files inside your project.
+
+![local-plugin-structure](./../assets/local-plugin-structure.png)
+
+```
+.
+├── sw-plugins                  <- Local folder for developing and overriding plugins
+│   ├── local-plugins.json      <- Config file to manage the order of plugins
+│   ├── cool-promotion-plugin   <- your local plugin directory, maybe more of them
+│   │   ├── config.json         <- plugin configuration
+│   │   └── *.vue               <- vue files for your plugin
+│   ├── another-local-plugin
+│   │   ├── config.json
+│   │   └── *.vue
+```
+
+so in `local-plugins.json` we're activating our new plugin
+
+```json
+{
+  "cool-promotion-plugin": true
+}
+```
+
+inside `cool-promotion-plugin/config.json` we need to put
+
+- slot, in which we want to inject
+- vue file, which should be displayed there
+
+In the single plugin, we may inject in multiple plugin slots. Let's inject our `main.vue` file into `top-header-after` slot.
+
+```json{4,5}
+{
+  "slots": [
+    {
+      "name": "top-header-after",
```

Comment:
Can one plugin inject multiple components into the same slot or would they have to be composited into a single one?

Reply:
no, it always should be a single component at the end as it some stage we'll add also context prop to API, so, for example, some slots will get current category or product as prop.

Why would you want to add two components? Could you show some use case for this?

Reply:
I'm not sure, that was just a question that came up to me. I think it's always possible to create a "container" component that just renders both of the components you want to add instead of adding them individually.

Reply:
I believe it's something we can consider in future when we'll have a use case for that, I see some dangerous area here:
* how to display these blocks? In a row, column?
* Where to put slot content? In first? Second? Both?

When there is a single entry point for plugin developer he's able to determine all of that and get the desired result quickly.
I'm not saying we shouldn't think about something like this, just more like find some use cases for that and figure out how the best approach would be :) But it's another issue I believe.

Reply:
I agree, let's live it at this for now.